src/metamodels/crosscat.py

```diff
@@ -220,15 +227,62 @@ class CrosscatMetamodel(metamodel.IBayesDBMetamodel):
 
     Internally, the Crosscat metamodel adds SQL tables to the database
     with names that begin with ``bayesdb_crosscat_``.
+
+    `crosscat` can be a crosscat engine instance or constructor. Prefer the
+    latter case, as that way bayeslite stochasticity is passed to the
+    constructor `seed` argument when engines are constructed for crosscat
+    queries (which at the time of writing happens every time bayeslite makes a
+    query.) The 'crosscat as instance' case is kept for backwards
+    compatibility.
+
+    Arguments to the constructor can be passed by ccargs and cckwargs. If using
+    a crosscat MultiprocessingEngine, the process pool must be passed as a
+    'pool' argument via cckwargs. It would not do for a new process pool to be
+    created every time a new query is made.
+
     """
 
-    def __init__(self, crosscat, subsample=None):
-        if subsample is None:
-            subsample = False
-        self._crosscat = crosscat
-        self._subsample = subsample
+    def __init__(self, crosscat, subsample=None, ccargs=None,
+                 cckwargs=None):
+        if isinstance(crosscat, EngineTemplate):
```

Comment:
No more dynamic decisions on the basis of `isinstance` please, unless actually operating on a data structure that is sensibly a union type (e.g., a json tree).